bundles/org.openhab.binding.heliosventilation/src/main/java/org/openhab/binding/heliosventilation/internal/HeliosVentilationHandler.java

```diff
@@ -0,0 +1,416 @@
+
+                out.write(txFrame);
+                // after each frame we have to wait.
+                // 30 ms is taken from what we roughly see the original remote control is doing
+                Thread.sleep(30);
+            }
+        } catch (IOException e) {
+            // in case we cannot write the connection is somehow broken, let's officially disconnect
+            disconnect();
+            connect();
+        } catch (InterruptedException e) {
+            // ignore if we got interrupted
+        }
+    }
+
+    /**
+     * Check connection status
+     *
+     * @return true if currently connected
+     */
+    private boolean isConnected() {
+        return serialPort != null && inputStream != null && outputStream != null;
+    }
+
+    @Override
+    public void serialEvent(SerialPortEvent event) {
+        switch (event.getEventType()) {
+            case SerialPortEvent.DATA_AVAILABLE:
+                // we get here if data has been received
+                try {
+                    try {
+                        // wait a little to increase the likelihood that the complete frame is already buffered.
+                        // this improves the robustness for RS485/USB converters which sometimes duplicate bytes
+                        // otherwise
+                        Thread.sleep(5);
+                    } catch (InterruptedException e) {
+                        // ignore interruption
+                    }
+                    byte frame[] = { 0, 0, 0, 0, 0, 0 };
+                    InputStream in = inputStream;
+                    if (in != null) {
+                        do {
+                            int cnt = 0;
+                            int c;
+                            do {
+                                c = in.read();
+                                if (cnt > 0 || c == 0x01) {
+                                    frame[cnt] = (byte) c;
+                                    cnt++;
+                                }
+
+                                if (cnt < 6 && in.available() < 1) {
+                                    // frame not yet complete but no input available, let's wait a little to merge
+                                    // interrupted transmissions
+                                    try {
+                                        Thread.sleep(10);
+                                    } catch (InterruptedException e) {
+                                        // ignore interruption
+                                    }
+                                }
+                            } while (in.available() > 0 && (c != -1) && cnt < 6);
+                            int sum = checksum(frame);
+                            if (sum == (frame[5] & 0xff)) {
+                                logger.trace("HeliosVentilation: Read from serial port: {}",
+                                        String.format("%02x %02x %02x %02x", frame[1], frame[2], frame[3], frame[4]));
```

Comment:
String formatting can be a bit computation heavy, so only run that code when needed.
```suggestion
                                if(logger.isTraceEnabled()){
                                    logger.trace("HeliosVentilation: Read from serial port: {}",
                                            String.format("%02x %02x %02x %02x", frame[1], frame[2], frame[3], frame[4]));
                                }
```